src/store/slices/auth/auth.thunks.ts

```diff
@@ -0,0 +1,45 @@
+import {registerUserUrl, loginUserUrl, getUserUrl} from '@/constants/api';
+import axiosInstance from '@/services/axiosInstance';
+import {createAsyncThunk} from '@reduxjs/toolkit';
+import {AxiosError, AxiosResponse} from 'axios';
+import {SignInSchemaType} from '@/constants/signIn.validation';
+import {SignUpSchemaTypeAPI} from '@/constants/signUp.validation';
+
+export const registerUser = createAsyncThunk(
+  'auth/registerUser',
+  async (userData: SignUpSchemaTypeAPI): Promise<any> => {
+    try {
+      const response: AxiosResponse = await axiosInstance.post(registerUserUrl, userData);
```

Comment:
Probably you can try return the destructured data? To avoid `AxiosResponse` in each API?

Reply:
So I renamed response to data. Because I use interceptor and get only data, not the whole response. And removed AxiosResponse type.

Reply:
But I suggest you using more sematic name not `data` everywhere to avoid a huge amount of `data` word. It is goin to be confusing it the future.

Reply:
Agree. Now I return like this
```
return await axiosInstance.post(registerUserUrl, userData);
```